tests/utils/test_dataset_combine.py

```diff
@@ -231,3 +232,53 @@ def test_combine_minari_version_specifiers(specifier_intersection, version_speci
     intersection = combine_minari_version_specifiers(version_specifiers)
 
     assert specifier_intersection == intersection
+
+
+# in the future, if the logic of save metadata of combined dataset changes, this should be changed as well
+def test_combine_dataset_with_different_metadata():
+    n_data = 2
+    dataset_list = []
+    for i in range(n_data):
+        dataset_id = f"cartpole-test-{i}-v0"
+        env = gym.make("CartPole-v1", max_episode_steps=500)
+        env = DataCollectorV0(env)
+        env.reset(seed=42)
+        for episode in range(5):
+            terminated = False
+            truncated = False
+            while not terminated and not truncated:
+                action = env.action_space.sample()  # User-defined policy function
+                _, _, terminated, truncated, _ = env.step(action)
+            env.reset()
+
+        # Create Minari dataset and store locally
+        dataset = minari.create_dataset_from_collector_env(
+            dataset_id=dataset_id,
+            collector_env=env,
+            algorithm_name="random_policy" + str(i),
+            code_permalink="https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
+            + str(i),
+            author="WillDudley" + str(i),
+            author_email="wdudley@farama.org" + str(i),
+        )
+        assert isinstance(dataset, MinariDataset)
+        env.close()
+        dataset_list.append(dataset)
+
+    combined_dataset = combine_datasets(
+        dataset_list, new_dataset_id="cartpole-combined-test-v0"
+    )
+    with h5py.File(combined_dataset.spec.data_path) as dt_file:
+        assert dt_file.attrs["algorithm_name"] == "random_policy" + str(n_data - 1)
+        _final_code_link = (
+            "https://github.com/Farama-Foundation/Minari/blob/f095bfe07f8dc6642082599e07779ec1dd9b2667/tutorials/LocalStorage/local_storage.py"
+            + str(n_data - 1)
+        )
+        assert dt_file.attrs["code_permalink"] == _final_code_link
+        assert dt_file.attrs["author"] == "WillDudley" + str(n_data - 1)
+        assert dt_file.attrs["author_email"] == "wdudley@farama.org" + str(n_data - 1)
```

Comment:
btw, this behavior is a little weird in combine dataset (you just save the attributes of the last)
Anyway, it is fixed in https://github.com/Farama-Foundation/Minari/pull/133 that we will merge after, so it is okay for now

Reply:
Ah, yes, I don't think only saving the attribute of last is a good choice, but it's just to keep same with `combine_dataset()` implementation. Definitely it will be changed in the future, but this PR is just to save the fogortten metadata `code_permalink` and `algorithm_name`.